sympy/polys/tests/test_poly_hypothesis.py

```diff
@@ -0,0 +1,22 @@
+from hypothesis import given
+from hypothesis import strategies as st
+from sympy.abc import x
+from sympy.polys.polytools import Poly
+
+
+@given(
+    coefficients1=st.lists(st.integers()),
+    coefficients2=st.lists(st.integers()).filter(lambda x: not all(i == 0 for i in x)),
```

Comment:
It may be better to sample on the cases where the first element of list is nonzero.

Reply:
I'm not understanding. This is just to remove the zero division case. We don't want the second polynomial to be 0, i.e., we don't want the list of coefficients to just be a bunch of 0's. This is the only case this filter removes.

Reply:
You don’t always have to test that every coefficients are zero, but only the leading coefficient is not zero is sufficient.

This leaves lots of cases like 0012, … which is same as 12, …, if you think of decimals

Reply:
It’s still a matter of semantics. 
I’d like to define the nonzero polynomials as something like leading coefficient is not zero though.
We may have other sort of invariant checking with only the leading coefficients.

Reply:
In the future I think we'll want strategies that generate Poly objects directly, and for those, we should definitely try to do this (we have to be careful that it's still possible to generate 0 itself, though).